bin/setup.dart

```diff
@@ -0,0 +1,135 @@
+    final downloadUrl = cronetBinaryUrl + fileName;
+    logger.stdout(downloadUrl);
+    try {
+      final request = await HttpClient().getUrl(Uri.parse(downloadUrl));
+      final response = await request.close();
+      final fileSink = File(fileName).openWrite();
+      await response.pipe(fileSink);
+      await fileSink.flush();
+      await fileSink.close();
+    } catch (error) {
+      Exception("Can't download. Check your network connection!");
+    }
+
+    placeBinaries(platform, fileName);
+  } else {
+    logger.stdout('${ansi.yellow}Cronet $platform is already available.'
+        ' No need to download.${ansi.none}');
+  }
+}
+
+/// Builds wrapper from source for the current platform.
+void buildWrapper() {
+  final logger = Logger.standard();
+  final ansi = Ansi(Ansi.terminalSupportsAnsi);
+  final wrapperPath = wrapperSourcePath();
+  final pwd = Directory.current;
+  Directory.current = Directory(wrapperPath);
+  logger.stdout('Building Wrapper...');
+  try {
+    final result = Process.runSync(
+        'cmake', ['CMakeLists.txt', '-B', 'out/${Platform.operatingSystem}']);
+    print(result.stdout);
+    print(result.stderr);
+  } catch (error) {
+    Directory.current = pwd;
+    logger.stdout("${ansi.red}Build failed.${ansi.none}");
+    if (Platform.isWindows) {
+      logger.stdout(
+          'Open ${ansi.yellow}x64 Native Tools Command Prompt for VS 2019.${ansi.none} Then run:\n');
+      logger.stdout('cd ${pwd.path}\ndart run cronet:setup build');
+    }
+    return;
+  }
+  var result =
+      Process.runSync('cmake', ['--build', 'out/${Platform.operatingSystem}']);
+  print(result.stdout);
+  print(result.stderr);
+  if (result.exitCode != 0) return;
+  Directory.current = pwd;
+  final moveLocation = '$binaryStorageDir${Platform.operatingSystem}64';
+  Directory(moveLocation).createSync(recursive: true);
+  final buildOutputPath = Platform.isLinux
+      ? '$wrapperPath/out/${Platform.operatingSystem}/${getWrapperName()}'
+      : '$wrapperPath\\out\\${Platform.operatingSystem}\\Debug\\${getWrapperName()}';
+  File(buildOutputPath).copySync('$moveLocation/${getWrapperName()}');
+  logger.stdout(
+      '${ansi.green}Wrapper moved to $moveLocation. Success!${ansi.none}');
+  return;
+}
+
+Future<void> main(List<String> args) async {
+  const docStr = """
+dart run cronet:setup [option]
+Downloads the cronet binaries.\n
+clean\tClean downloaded or built binaries.
+build\tBuilds the wrapper. Requires cmake.
+  """;
+  final logger = Logger.standard();
+  if (args.length > 1) {
+    logger.stderr('Expected 1 argument only.');
+    logger.stdout(docStr);
+  } else if (args.contains('-h')) {
+    logger.stdout(docStr);
+  } else if (args.contains('clean')) {
+    logger.stdout('cleaning...');
+    Directory(binaryStorageDir).deleteSync(recursive: true);
+    logger.stdout('Done!');
+  } else if (args.contains('build')) {
+    buildWrapper();
+  } else {
+    for (final platform in validPlatforms) {
```

Comment:
Platform.isWindows Platform.isLinux can tell you on which platform we are, we don't need to download the others right?